src/Statistics/views.py

```diff
@@ -45,4 +53,342 @@ def durationPlay(request):
+            data1.append(el1)
+            data2.append(el2)
+            idx += 1
+
+    elif len(tableTitlesData) == 7:
+        for el1, el2, el3 in data:
+            pdf.cell(297, 7, str(idx).ljust(tableContentData[0]) + str(el1).ljust(tableContentData[1]) +
+                     str(el2).ljust(tableContentData[2]) + str(el3).rjust(tableContentData[3]), ln=1)
+            data1.append(el1)
+            data2.append(el2)
+            data3.append(el3)
+            idx += 1
+
+    return data1, data2, data3, pdf
+
+
+def addContentToMyReport(pdf, data, tableContentData, title, tableTitlesData, xLabel, yLabel, id_):
+    data1, data2, data3, pdf = addToPDF(pdf, data, title, tableTitlesData, tableContentData)
+    x = [str(i) for i in range(1, len(data1) + 1)]
+
+    if len(tableTitlesData) == 3:
+        createBarPlot(x, data2, title, xLabel, yLabel, id_)
+    elif len(tableTitlesData) == 7:
+        createBarPlot(x, data3, title, xLabel, yLabel, id_)
+
+    pdf.image('image' + str(id_) + '.jpg', x=23.5, w=250, h=176)
+
+
+def createPiePlot(title, data, counts, id_):
+    maxi = max(counts)
+    idx = counts.index(maxi)
+    values = np.array(counts)
+
+    explode = [0 for _ in range(len(counts))]
+    explode[idx] = 0.1
+    colors = findNDifferentColors(len(counts))
+
+    plt.pie(values, colors=colors, explode=explode, shadow=False, startangle=140)
+    plt.title(title)
+    plt.axis('equal')
+
+    labels = ['id:' f'{l} - {(s / sum(counts) * 100):0.2f}%' for l, s in zip(data, counts)]
+    plt.legend(bbox_to_anchor=(0.8, 0), loc='lower left', labels=labels)
+    plt.savefig("image" + str(id_) + ".jpg", dpi=300)
+    plt.close()
+    plt.clf()
+
+
+def createBarPlot(xValues, yValues, title, xLabel, yLabel, id_):
+    plt.bar(xValues, yValues, color='#21a25b', width=0.6)
+    plt.title(title)
+    plt.xlabel(xLabel)
+    plt.ylabel(yLabel)
+
+    for i in range(len(xValues)):
+        plt.text(i, yValues[i], yValues[i], ha='center')
+
+    plt.savefig("image" + str(id_) + ".jpg", dpi=300)
+    plt.close()
+    plt.clf()
+
+
+# create different reports
+
+
+def myReport(request):
+    user = request.user
+    fiveBestSongs = findBestFiveSongs(findMostPopularUserSongs(user))
+    fiveBestArtists = findBestFiveSongs(findMostPopularUserArtists(user))
+    fiveBestGenres = findBestFiveSongs(findMostPopularUserGenres(user))
+
+    pdf = createPDF()
+
+    tableTitlesData = ['Id', 7, 'Title', 50, 'Artist', 25, 'Count']
+    tableContentData = [10, 60, 26, 10]
+    addContentToMyReport(pdf, fiveBestSongs, tableContentData, 'Number of plays of my 5 best songs', tableTitlesData, 'Song ID', 'Number', 5)
+
+    tableTitlesData = ['Id', 7, 'Artist', 75, 'Count']
+    tableContentData = [10, 92, 0]
+    addContentToMyReport(pdf, fiveBestArtists, tableContentData, 'Number of plays of my best 5 artist\'s song', tableTitlesData, 'Artist', 'Number', 6)
+
```

Comment:
_tableTitlesData_ has a length of 5 then is passed to **addContentToMyReport()**

Reply:
fixed this issue